### PR TITLE
[IMP] l10n_fr: Update COA and account groups

### PR DESCRIPTION
**Pull request by FlorianGilbert**
The aim of this commit is updating the COA and the account groups for the new legislation in France.

task-4040854

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by malb-odoo — COMMENTED** [review]
The forward port will be painful i suppose we did the translation of the coa in +17.0 ahah

Nothing to add for the COA, just for the account group there is a lot of space after the double quote and i see that sometimes you changed that but not all the time. 

No translation needed since it's already in french :vomiting_face: but you will need to do it when the coa will be in english :grimacing:

**Review by malb-odoo — COMMENTED** [review]
LGTM Thankssss for this painful job :partying_face:

**Review by malb-odoo — APPROVED** [review]

**Review by gawa-odoo — COMMENTED** [review]
Updating the module with your new code gets a traceback `For external id l10n_fr.pcg_394 when trying to create/update a record of model account.account.template found record of different model account.group.template (20302)`

File: addons/l10n_fr/data/account.account.template.csv [hunk 1]
```diff
@@ -1,233 +1,180 @@
...
-"pcg_1398","Autres subventions d'investissement (même ventilation que celle du compte 1391)","139800","equity","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_1423","Provisions reconstitution des gisements miniers et pétroliers","142300","liability_non_current","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_1424","Provisions pour investissement (participation des salariés)","142400","liability_non_current","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_1431","Provisions réglementées relatives aux stocks - Hausse de prix","143100","liability_non_current","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_1432","Provisions réglementées relatives aux stocks - Fluctuation des cours","143200","liability_non_current","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_144","Provisions réglementées relatives aux autres éléments de l'actif","144000","liability_non_current","l10n_fr.l10n_fr_pcg_chart_template","","False"
+"pcg_131","Subventions d'équipement","131000","equity","l10n_fr.l10n_fr_pcg_chart_template","","False"
+"pcg_139","Subventions d'investissement inscrites au compte de résultat","138000","equity","l10n_fr.l10n_fr_pcg_chart_template","","False"
```
Comment (gawa-odoo) [inline addons/l10n_fr/data/account.account.template.csv:29]: I don't know if it's supposed to be 138 or 139, but the mismatch is weird.

File: addons/l10n_fr/data/account.account.template.csv [hunk 2]
```diff
@@ -383,22 +307,26 @@
...
-"pcg_4771","Différences de conversion - Passif - Augmentation des créances","477100","liability_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
-"pcg_4772","Différences de conversion - Passif - Diminution des dettes","477200","liability_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
-"pcg_4778","Différences de conversion - Passif - Différences compensées par couverture de change","477800","liability_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
+"pcg_472","Comptes d'attente","472000","asset_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
+"pcg_473","Comptes d'attente","473000","asset_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
+"pcg_4741","Différences d'évaluation sur instruments financiers à terme - Actif","474100","asset_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
+"pcg_4742","Différences d'évaluation sur jetons détenus - Actif","474200","asset_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
+"pcg_4751","Différences d'évaluation sur instruments financiers à terme - Passif","475100","asset_current","l10n_fr.l10n_fr_pcg_chart_template","","True"
```
Comment (gawa-odoo) [inline addons/l10n_fr/data/account.account.template.csv:317]: Passif that is an asset? 
Same idea for some accounts below

File: addons/l10n_fr/data/account.account.template.csv [hunk 3]
```diff
@@ -520,7 +437,7 @@
 "pcg_6235","Primes","623500","expense","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_operating","False"
 "pcg_6236","Catalogues et imprimés","623600","expense","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_operating","False"
 "pcg_6237","Publications","623700","expense","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_operating","False"
-"pcg_6238","Publicité, publications, relations publiques - Divers (pourboires, dons courants...)","623800","expense","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_operating","False"
+"pcg_6238","Divers (pourboires, dons courants)","623800","expense","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_investing","False"
```
Comment (gawa-odoo) [inline addons/l10n_fr/data/account.account.template.csv:440]: Really, we put it in investing? Isn't it weird? I know there are some gifts, but this feels like small donations and not investment

File: addons/l10n_fr/data/account.account.template.csv [hunk 4]
```diff
@@ -1,233 +1,180 @@
...
-"pcg_2814","Amortissements des immobilisations corporelles - Constructions sur sol d'autrui (même ventilation que celle du compte 214)","281400","asset_fixed","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_2815","Amortissements des immobilisations corporelles - Installations matériel et outillage industriels (même ventilation que celle du compte 215)","281500","asset_fixed","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_2818","Amortissements des autres immobilisations corporelles (même ventilation que celle du compte 218)","281800","asset_fixed","l10n_fr.l10n_fr_pcg_chart_template","","False"
-"pcg_28187"," Amortissement du mali de fusion sur actifs corporels","281870","asset_fixed","l10n_fr.l10n_fr_pcg_chart_template","","False"
+"pcg_2801","Frais d'établissement (même ventilation que celle du compte 201)","280100","asset_fixed","l10n_fr.l10n_fr_pcg_chart_template","","False"
+"pcg_2803","Frais de développement","280300","asset_fixed","l10n_fr.l10n_fr_pcg_chart_template","","False"
```
Comment (gawa-odoo) [inline addons/l10n_fr/data/account.account.template.csv:152]: I will say that the fact that we don't make any diff in the name between 203, 2803, 2903 feels a bit weird.
Nothing really wrong, as the group defines it, I'm just making the remark

**Comment by gawa-odoo** [comment]
@robodoo r+

File: addons/l10n_fr/data/account.account.template.csv [hunk 5]
```diff
@@ -683,16 +589,26 @@
...
-"pcg_758","Produits divers de gestion courante","758000","income","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_operating","False"
+"pcg_757","Produits des cessions d’immobilisations incorporelles et corporelles","757000","income","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_investing","False"
+"pcg_7581","Dédits et pénalités perçus sur achats et ventes","7581100","income","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_investing","False"
+"pcg_7582","Libéralités reçues","758200","income","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_investing","False"
+"pcg_7583","Rentrées sur créances amorties","758300","income","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_investing","False"
+"pcg_7584","Dégrèvements d’impôts autres qu’impôts sur les bénéfices","758400","income","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_investing","False"
+"pcg_7585","Bonis provenant de clauses d’indexation","778100","income","l10n_fr.l10n_fr_pcg_chart_template","account.account_tag_investing","False"
```
Comment (remi-filament) [inline addons/l10n_fr/data/account.account.template.csv:608]: Should not External ID be pcg_7781 here to keep coherency with account number ?

**Review by remi-filament — COMMENTED** [review]
Thanks @FlorianGilbert for this impressive rework !
I have noticed 2 external IDs on accounts that do not match account number, do you think it is worth aligning those ?
Best Regards

**Comment by alexis-via** [comment]
Glad to see that the big cleanup I made in the French chart of account in l10n_fr_oca has now been made in l10n_fr ! It was really needed.

**Comment by risto42** [comment]
I sincerely suggest backing out this changeset opting for one more useable. Notably, for the accounts that are no longer in the 2025 PCG, they should probably only be deprecated as companies have some/most/all of 2025 to organise their migration for the fiscal year 2025.  But note that up through FY2024 these accounts should not be affected at all, but such is the case with this commit.

**Comment by gawa-odoo** [comment]
> Thanks @FlorianGilbert for this impressive rework ! I have noticed 2 external IDs on accounts that do not match account number, do you think it is worth aligning those ? Best Regards

I opened a PR to fix it here https://github.com/odoo/odoo/pull/191826 (and other typos I found)

**Comment by gawa-odoo** [comment]
> I sincerely suggest backing out this changeset opting for one more useable. Notably, for the accounts that are no longer in the 2025 PCG, they should probably only be deprecated as companies have some/most/all of 2025 to organise their migration for the fiscal year 2025. But note that up through FY2024 these accounts should not be affected at all, but such is the case with this commit.

Hello @risto42 
Can you explain what your issue with these changes are?
Here we only change the templates. So it will only apply to new companies (and dbs). The accounts and groups already created won't be deleted.
From 17.0 onwards, only if you reload the CoA, will you get the new accounts/groups. We still won't delete old ones.

**Comment by risto42** [comment]
New to Odoo companies, not only newly created companies.

If migrating from a legacy system (currently with FY<= 2024) to a new database initialised with:
 `--load-language=fr_FR -i base,l10n_fr', there are current accounts either moved or deleted making it a royal PITA to deal with upon short notice.

Furthermore, we have plenty of time to organise the PCG migration **after** the accounting system migration.

In any event, as indicated by Lefebvre:
```
Modalités d’établissement des premiers états financiers

Les nouvelles dispositions du projet de règlement n°2022-06 seront appliquées pour la présentation des états financiers pour le 1er exercice clos à compter du 1er janvier 2025, et impacteront également les informations portées dans la colonne N-1, induisant ainsi des reclassements pour les rendre compatibles avec le nouveau format de présentation des états financiers.

Il est à noter que pour la 1ère présentation du tableau des flux de trésorerie, la colonne N-1 ne sera pas obligatoire.
```
In other words, the migration to the new PCG standard needs to be completed typically prior to 2026 when the accounts are ultimately declared to the fiscal authorities (SIE).

And as I noted elsewhere, ideally, the template should simply be versioned, and use in the COA could involve a start_date/end_date in concert with replaced_by for the affected accounts. This making for an easier to manage PCG migration _when_ the company and accountants feel it's time to do so.
